src/timely-util/src/operator.rs

```diff
@@ -411,29 +413,28 @@ where
 
     fn explode_one<D2, R2, L>(&self, mut logic: L) -> Collection<G, D2, <R2 as Multiply<R>>::Output>
     where
-        D2: Data + Ord,
+        D2: differential_dataflow::Data,
         R2: Semigroup + Multiply<R>,
         <R2 as Multiply<R>>::Output: Data + Semigroup,
         L: FnMut(D1) -> (D2, R2) + 'static,
+        G::Timestamp: Lattice,
     {
-        let mut buffer = Vec::new();
-        let mut out = Vec::new();
         self.inner
             .unary(Pipeline, "ExplodeOne", move |_, _| {
+                let mut buffer = Vec::new();
                 move |input, output| {
+                    let mut out = ConsolidateBuffer::new(output, 0);
                     input.for_each(|time, data| {
                         data.swap(&mut buffer);
-                        out.extend(buffer.drain(..).map(|(x, t, d)| {
+                        let exploded = buffer.drain(..).map(|(x, t, d)| {
                             let (x, d2) = logic(x);
                             (x, t, d2.multiply(&d))
-                        }));
-                        differential_dataflow::consolidation::consolidate_updates(&mut out);
-                        if out.len() < out.capacity() / 2 {
-                            output.session(&time).give_iterator(out.drain(..));
-                        } else {
-                            output.session(&time).give_container(&mut out);
+                        });
+                        for item in exploded {
+                            out.give(&time, item);
                         }
                     });
+                    out.flush();
```

Comment:
`ConsolidateBuffer` flushes on drop, so there's no need to call `flush` explicitly.

Reply:
Indeed, I removed this to make the use in `explode_one` more idiomatic and to improve the calling code style even further, added a `give_iterator` function to `ConsolidateBuffer`.